controllers/controller.go

```diff
@@ -325,6 +327,20 @@ func (r *EtcdadmClusterReconciler) reconcile(ctx context.Context, etcdCluster *e
 				delete(etcdCluster.Annotations, etcdv1.UpgradeInProgressAnnotation)
 			}
 		}
+
+		// If the ETCD nodes have performed a rollout, the etcd client certs on the CP nodes need to be renewed.
+		// We mark the condition EtcdCertificatesAvailable False in the rollout case and check for its value.
+		// The default case is hit right after ScaleUp of ETCD nodes is completed and before the first CP comes up.
+		// If EtcdCertificatesAvailable is False, this means we need to update the certs.
+		// EtcdCertificatesAvailable is set to True once the certs are updated.
+		if conditions.Has(ep.EC, etcdv1.EtcdCertificatesAvailableCondition) &&
+			conditions.IsFalse(ep.EC, etcdv1.EtcdCertificatesAvailableCondition) {
+			log.Info("Updating Etcd client certs")
+			if err := r.generateCAandClientCertSecrets(ctx, cluster, etcdCluster); err != nil {
```

Comment:
the actual external etcd CA cert + key pair are not regenerated after the etcd machines are rolled out right?

Reply:
So the ETCD certs on the ETCD nodes do get updated but thats with etcdadm, its not related to what we do here.

That was working, the problem was only that the secret on the cluster was not updated and thereby the cert on the CP was not updated.